src/analysis/convex/stone_separation.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import analysis.convex.join
+
+/-!
+# Stone's separation theorem
+
+This file prove Stone's separation theorem. This tells us that any two disjoint convex sets can be
+separated by a convex set whose complement is also convex.
+
+It can be seen as a "cheap" version of the geometric Hahn-Banach theorem, in the sense that the
+separating set in Stone's separation corresponds to the halfspace in geometric Hahn-Banach, but in a
+much broader class of spaces (any vector space, instead of real normed spaces).
+-/
+
+open set
+open_locale big_operators
+
+variables {𝕜 E ι : Type*} [linear_ordered_field 𝕜] [add_comm_group E] [module 𝕜 E] {w x y z : E}
+  {s t : set E}
+
+/-- In a tetrahedron, any segment joining opposite edges passes through any triangle whose base is
+another edge. -/
```

Comment:
How about a few labels. Maybe something like:
```suggestion
/-- In a tetrahedron `p, q, x, y`, any segment `u, v` joining opposite edges, passes through any triangle `p, q, z` with base consisting a third edge, and vertex `z` lying on a fourth edge (opposite). -/
```
(it took me a few moments to spot what was what).